### DISCUSSION TITLE
Tracktable 1.7.0

### DISCUSSION THREAD

**Discussion by atwilso** [sandialabs/tracktable#6] [Announcements]
Welcome back!

Most of the work in this release is in the build and test infrastructure to make our development practices more sustainable.  We've added the following user-facing features:

- Maps in Folium can now be created with the `attr` and `crs` 
  parameters.  The `attr` argument is used to pass an attribution string
  for custom tile sets.  The `crs` argument names a coordinate transform
  to project geographical points into pixel coordinates and back.  The
  functions `tracktable.render.render_trajectories.render_trajectories()` 
  and `tracktable.render.render_heatmap.render_heatmap()` both support these
  new arguments.

- Trajectory maps in Folium can be animated.  See the arguments `animate`,
  `anim_display_update_interval`, `anim_trail_duration`, and `use_markers`
  in `tracktable.render.render_trajectories.render_trajectories()`.
  
- New function `tracktable.rw.load.load_trajectories()` that will load a
  `.traj`, `.csv`, or `.tsv` file and assemble points into trajectories.
  An optional flag will cause it to return points instead of trajectories.
  Examples that load trajectories or points have been updated to use this
  feature.

- Sea ports and airports can be rendered into maps for both the Cartopy
  and Folium back ends.

- Docs and data have been separated out into their own repositories
  implemented as submodules. Note that you will need to `git clone --recursive` 
  when cloning the source code.  The `tracktable-data` package should be installed
  as a dependency when you install with `pip` or from conda-forge.

Full details are available in the [https://tracktable.readthedocs.io](documentation).

<hr /><em>This discussion was created from the release <a href='https://github.com/sandialabs/tracktable/releases/tag/v1.7.0'>Tracktable 1.7.0</a>.</em>

**Comment by mark-rintoul** [comment]
Nice to see the latest upgrades!

**Reply by mark-rintoul** [reply]
And for the record, I had trouble compiling it with gcc 11.4.0.  Line 10827 in tracktable/ThirdParty/catch2.hpp reads

`static constexpr std::size_t sigStackSize = 32768 >= MINSIGSTKSZ ? 32768 : MINSIGSTKSZ;`

and should read

`static constexpr std::size_t sigStackSize = 32768;`

thanks to some new changes in gcc (I believe).  This seems to be a known issue in various compilations.

**Reply by atwilso** [reply]
Thanks, Danny!  I've gone ahead and fixed that in the patches for 1.7.1 (coming out very soon).

We miss you and Michael a lot.

**Comment by mark-rintoul** [comment]
Still trying to deprogram myself from being a Sandian.  I needed some
change to kickstart me, and boy did I get some.  All is still currently
chaos, but I'm slowly adjusting.  I sure wish I'd spent more time talking
databases with you before I took this job!

           Danny
